mvpa2/datasets/miscfx.py

```diff
@@ -416,6 +418,71 @@ def cl_stats(axis, u, name1, name2):
     return s
 
 
+@datasetmethod
+def to_table(ds, sa=[], fa=[], height=100, width=10, n_digits=6):
+    height = min(ds.shape[0], height)
+    width = min(ds.shape[1], width)
+
+    def create_attr_panel(ds_attr, keys, length):
+        panel = []
+        panel.append(keys) #  add header
+        for i in range(length):
+            panel.append([ds_attr[key][i] for key in keys])
+        return panel
+
+    # create empty panel if there are no attributes
+    if len(ds.sa.keys()) == 0:
+        ds.sa[None] = [None]*height
+    if len(ds.fa.keys()) == 0:
+        ds.fa[None] = [None]*width
+
+    if sa == []:
+        sa = ds.sa.keys()
+    elif isinstance(sa, basestring):
+        sa = [sa]
+
+    if fa == []:
+        fa = ds.fa.keys()
+    elif isinstance(fa, basestring):
+        fa = [fa]
+
+    sa_panel = create_attr_panel(ds.sa, sa, height)
+    fa_panel = np.array(create_attr_panel(ds.fa, fa, width)).transpose()
+    vals_panel = np.vstack([ds[i].samples[0][0:width] for i in range(height)])
+    void = np.vstack([['  ']*max(len(sa), 1)] * max(len(fa), 1))
+
+    # add empty row and column, to align it with headers
+    vals_panel = np.hstack((np.array([" "*n_digits]*height).reshape(-1, 1),
+                            vals_panel))
+    vals_panel = np.vstack(([""]*(width+1), vals_panel))
+
+    return np.vstack((np.hstack((void, fa_panel)),
+                              np.hstack((sa_panel, vals_panel))))
+
+@datasetmethod
+def prt(ds, sa=[], fa=[], height=50, width=12, n_digits=5):
+    """Prints dataset object in table form
+
+    Parameters
+    ----------
+    sa : list or string, optional
+      Names of sample attributes to be displayed. Empty list will display all
+    fa : list or string, optional
+      Names of feature attributes to be displayed, Empty list will display all
+    height : int, optional
+      Number of rows printed
+    width : int, optional
+      Number of columns printed
+    ndigits : int, optional
+      Number of digits displayed in one cell
+    """
+    height = min(ds.shape[0], height)
+    width = min(ds.shape[1], width)
+    print
+    print tabulate(to_table(ds, sa, fa, height, width, n_digits))
+    print "Ommitted %s rows and %s columns" %(ds.shape[0]-height, ds.shape[1]-width)
```

Comment:
instead of straight print better add kwarg "out=sys.stdout", and then use `out.write()` and append '\n' at the end for each line.  That would allow to use this function to dump output anywhere and also to test easily (which needs to be done)